lib/errors/errors.go

```diff
@@ -126,3 +126,31 @@ type ErrorResourceUpdateConflict struct {
 func (e ErrorResourceUpdateConflict) Error() string {
 	return fmt.Sprintf("update conflict: '%s'", e.Identifier)
 }
+
+// UpdateErrorIdentifier modifies the supplied error to use the new resource
+// identifier.
+func UpdateErrorIdentifier(err error, id interface{}) error {
+	if err == nil {
+		return nil
+	}
+
+	switch e := err.(type) {
```

Comment:
Hm, why do we need to create a new var `e` and then re-assign to `err`?

Couldn't we just do `err.Identifier= id` for each of these?